packages/ui/src/helpers/accountSettings/validators/util.ts

```diff
@@ -0,0 +1,56 @@
+import { FieldValidator, InputEventType, ValidationMode } from '../../../types';
+
+/**
+ * `shouldValidate` determines whether validator should be run, based on validation mode,
+ * input event type, and whether it has been blurred yet.
+ */
+export const shouldValidate = (
+  validationMode: ValidationMode,
+  eventType: InputEventType,
+  hasBlurred: boolean
+) => {
+  switch (validationMode) {
+    case 'onBlur': {
+      // only run validator on blur event
+      return eventType === 'blur';
+    }
+    case 'onChange': {
+      // only run validtor on change event
+      return eventType === 'change';
+    }
+    case 'onTouched': {
+      /**
+       * run validator on first blur event, and then every subsequent
+       * blur/change event.
+       */
+      return eventType === 'blur' || hasBlurred;
+    }
+  }
+};
+
+/**
+ * `runFieldValidator` runs all validators, and returns error messages.
+ */
+export const runFieldValidators = (
+  field: string,
+  validators: FieldValidator[],
+  eventType: InputEventType,
+  hasBlurred: boolean
```

Comment:
`isDirty` is different, because `isDirty` just means that value has been modified even if input hasn't been blurred yet if I'm reading this correctly:

> Each [input](https://www.w3.org/TR/2011/WD-html5-20110525/the-input-element.html#the-input-element) element has a boolean dirty value flag. The [dirty value flag](https://www.w3.org/TR/2011/WD-html5-20110525/the-input-element.html#concept-input-value-dirty-flag) must be initially set to false when the element is created, and must be set to true whenever the user interacts with the control in a way that changes the [value](https://www.w3.org/TR/2011/WD-html5-20110525/association-of-controls-and-forms.html#concept-fe-value).

- https://www.w3.org/TR/2011/WD-html5-20110525/the-input-element.html